File: javascript/L3_getters_setters.js
class Student {
    #name;
    #age;

    get name() {
        return this.#name;
    }

    set name(newName) {
        this.#name = newName;
    }

    get age() {
        return this.#age;
    }

    set age(newAge) {
        if (newAge > 0) {
            this.#age = newAge;
        }
    }
}

const s1 = new Student();
s1.name = "Lohit";
s1.age = 21;

console.log(`Name: ${s1.name}`);
console.log(`Age: ${s1.age}`);

File: python/L3_getters_setters.py
class Student:
    def __init__(self, name, age):
        self._name = name
        self._age = age

    @property
    def name(self):
        return self._name

    @name.setter
    def name(self, new_name):
        self._name = new_name

    @property
    def age(self):
        return self._age

    @age.setter
    def age(self, new_age):
        if new_age > 0:
            self._age = new_age

s1 = Student("Lohit", 21)
print(f"Name: {s1.name}")
print(f"Age: {s1.age}")

s1.name = "Lohit P T"
s1.age = 22
print(f"New Name: {s1.name}")
print(f"New Age: {s1.age}")
